CMake/itkCompilerChecks.cmake

```diff
@@ -1,31 +1,33 @@
-# Minimum compiler version check: GCC >= 4.8
+# Minimum compiler version check: GCC
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
-  message(FATAL_ERROR "GCC 4.8 or later is required.")
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
+  message(FATAL_ERROR "GCC 5.1 or later is required.")
 endif ()
 
-# Minimum compiler version check: LLVM Clang >= 3.3
+# Minimum compiler version check: LLVM Clang
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
-  message(FATAL_ERROR "LLVM Clang 3.3 or later is required.")
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.4)
+  message(FATAL_ERROR "LLVM Clang 3.4 or later is required.")
 endif ()
 
-# Minimum compiler version check: Apple Clang >= 5.0 (Xcode 5.0)
+# Minimum compiler version check: Apple Clang (Xcode)
```

Comment:
@N-Dekker Please restore Xcode version in parens (to 5.1).  Xcode and AppleClang version numbers do not always match up (though they do here).

Reply:
@seanm Thanks, so what should it say exactly? `Apple Clang >= 5.1 (Xcode 5.1)` ?

Reply:
Yes, perfect. Looks redundant, but on my Mac for example I have Xcode 11.3.1 but it has AppleClang 11.0.0.  People are more familiar with their Xcode version than the actual compiler version.

Reply:
Thanks @seanm. Addressed by amend + force-push https://github.com/InsightSoftwareConsortium/ITK/compare/5539f556732dca021e8c9aebb4b12a52b61c428f..a35dcbb8d8469663fd58ef23765380c056f13eb2